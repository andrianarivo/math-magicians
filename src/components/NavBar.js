import { NavLink } from 'react-router-dom';
import styles from '../styles/NavBar.module.css';

export default function NavBar() {
  const links = [
    { path: '/', text: 'Home' },
    { path: 'calculator', text: 'Calculator' },
    { path: 'quote', text: 'Quotes' },
  ];
  return (
    <nav className={styles.nav}>
      <h2>Math Magicians</h2>
      <ul className={styles.ul}>
        {links.map((link) => (
          <li className={styles.li} key={link.path}>
            <NavLink to={link.path}>{link.text}</NavLink>
          </li>
        ))}
      </ul>
    </nav>
  );
}
