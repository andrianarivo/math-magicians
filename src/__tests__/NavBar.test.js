import renderer from 'react-test-renderer';
import { BrowserRouter as Router, Outlet } from 'react-router-dom';
import userEvent from '@testing-library/user-event';
import { render, screen } from '@testing-library/react';
import NavBar from '../components/NavBar';
import renderWithRoutes from '../utils/testUtils';

describe('testing NavBar component', () => {
  test('render', () => {
    const tree = renderer.create(
      <Router>
        <NavBar />
      </Router>
    );

    expect(tree).toMatchSnapshot();
  });

  test('user interaction', async () => {
    render(
      <Router>
        {renderWithRoutes(
          <>
            <NavBar />
            <Outlet />
          </>
        )}
      </Router>
    );

    await userEvent.click(screen.getByText('Calculator'));

    expect(screen.getByText('=')).toBeInTheDocument();
  });
});
