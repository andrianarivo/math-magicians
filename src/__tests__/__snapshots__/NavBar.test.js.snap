// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing NavBar component render 1`] = `
<nav
  className="nav"
>
  <h2>
    Math Magicians
  </h2>
  <ul
    className="ul"
  >
    <li
      className="li"
    >
      <a
        aria-current="page"
        className="active"
        href="/"
        onClick={[Function]}
      >
        Home
      </a>
    </li>
    <li
      className="li"
    >
      <a
        className=""
        href="/calculator"
        onClick={[Function]}
      >
        Calculator
      </a>
    </li>
    <li
      className="li"
    >
      <a
        className=""
        href="/quote"
        onClick={[Function]}
      >
        Quotes
      </a>
    </li>
  </ul>
</nav>
`;
